src/extensions/UserDefinedFormControllerExtension.php

```diff
@@ -24,6 +25,28 @@ class UserDefinedFormControllerExtension extends Extension
      */
     public function onBeforeInit()
     {
+        /** @var UserDefinedFormController $owner */
+        $owner = $this->owner;
+        $url = $owner->getRequest()->getURL();
+        // Start a clean session if the user visits the original form
+        // This should never run on the 'partial' or 'ping' URL
+        if (strpos($url, 'partial') !== 0 && !strpos($url, 'ping')) {
+            $startNew = true;
+            $session = $owner->getRequest()->getSession();
+            $id = $session->get(PartialSubmissionController::SESSION_KEY);
+            // Check if there is an existing partial submission
+            $existing = PartialFormSubmission::get()->byID((int)$id);
+            if ($existing) {
+                // Check if it has started yet, we need to start a new one, if it has started
+                $startNew = $existing->isStarted();
+            }
+
+            if ($startNew) {
+                $partialForm = PartialFormSubmission::create()->write();
```

Comment:
Can we set `Parent` and `UserDefinedForm` when we create a new PartialFormSubmission, so it doesn't set to AccordionItems automatically. I believe we can access by `$this->owner->getData()->ID` and `$this->owner->getData()->ClassName`